### ISSUE TITLE
Black textures, I am guessing vertex shader 2.0 vs 3.0 missmatch in Everquest

### ISSUE THREAD

**Issue by numinari** [doitsujin/dxvk#1618]
Less then fixing the issue which I don't really expect anything, I just want to know if my thoughts are correct with what is going on here.

https://community.amd.com/thread/252674 here is a link to what is going on and what i tried.

Essentially I get a black texture for most of the ground and walls in Eveverquest using the latest build 1.61 of dxvk. I was trying to use it because I don't get shadows under native directx which is no longer supported by windows or AMD. I am guessing its because my card doesnt support vertex shaders 2.0 and that is what you are using in your wrapper to the Vulkan calls. Or the wrapper cant translate vertex shader 3.0 calls and the engine doesnt have a setting for 2.0 that I can see. So I don't honestly think the issue is mainly on your end unless at some point 3.0 vertex shaders are implemented. Which I have really no clue about. I am just guessing totally at what is going on here.

### Software information
Windows 10, june 10 DX9 redist, DXvk 1.61 AMD radeon 5700 XT adrenaline 20.42 drivers. Everquest live .. latest client from daybreak games. Supposedly dx 9c according to the community.

Here's a link to the game forums thread were I am talking about what is happening: https://forums.daybreakgames.com/eq/index.php?threads/shadow-rendering-shadows-not-rendering.266000/#post-3907791

https://i.imgur.com/ixo0wUN.png is what the issue looks like

### System information
- Navi 10 AMD RX 5700 XT
- Driver: 20.42
- Wine version: none? windows 10?
- DXVK version: 1.61

### Apitrace file(s)
- Put a link here

### Log files
- d3d9.log:
[eqgame_d3d9.log](https://github.com/doitsujin/dxvk/files/4602362/eqgame_d3d9.log)

- d3d11.log:
none
- dxgi.log:

dxgi.log is empty

I am going to attempt to try changing some settings and stuff in dxvk and hopefully i can fool something into using the right shader, 2.0 i think...

Your wrapper is awesome. It is great to see this work being done as I don't think people with newer machines can even run d3d9 games in windows anymore. Much love!!! Thank you!

**Comment by Joshua-Ashton** [comment]
Please provide an apitrace.

**Comment by Joshua-Ashton** [comment]
You're also using DXVK 1.5.5, not 1.6.1 like you mentioned.

**Comment by numinari** [comment]
Oh let me reinstall 1.6.1 I tried both. I downloaded API trace last night so I will try and mess with that soon too.

**Comment by numinari** [comment]
I re-downloaded the latest build and got this output file.

[eqgame_d3d9.log](https://github.com/doitsujin/dxvk/files/4604188/eqgame_d3d9.log)

info:  Game: eqgame.exe
info:  DXVK: v1.6.1

info:  Enabled instance extensions:
info:    VK_KHR_get_surface_capabilities2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
warn:  D3D9: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
info:  AMD Radeon RX 5700 XT:
info:    Driver: 2.0.137
info:    Vulkan: 1.2.133

I also figured out how to get an api trace! I need to get one thats reasoonable in size I was having a little trouble with the program, I will upload a d3d9 and a dxgi one if I can soon!

**Comment by doitsujin** [comment]
Note that trace files compress fairly well with something like 7zip.

DXGI isn't needed here since the game is d3d9.

**Comment by numinari** [comment]
ok here is the d3d9 trace https://drive.google.com/file/d/191soNcvei4I2Jp-GuW424Lzyq72JZZMY/view?usp=sharing

you are right 7zip is pretty awesome!

**Comment by numinari** [comment]
hmm when I do a play back on my end I get an unhandled exeption and it doesn't show the in game trace, just the load screens, though I do get further in game -- attempting to further trouble shoot on my end

**Comment by numinari** [comment]
I used debugview to grab as much as I could and made a bug report over at the apitrace project. Thanks again, and if I ever get it sorted out I will repost a new trace.

**Comment by numinari** [comment]
An update from the API trace project the issue is Recursive locks on Everquest Live Win32 they are not easy to resolve. I suppose since this is so far only been produced under windows 10 not a wine or linux related issue. If you would like, I don't mind closing it under those grounds.

**Comment by duskDesigns** [comment]
I have this same issue with 'black' terrain in Everquest. I'm running Mint 19.2, 4.15.0-72-generic kernel, Nvidia RTX 2070 using Nvidia Driver 440.82.
Running game on -
Lutris with 5.6-2-x86_64
DXVK I've tried 1.5.5, 1.6 and 1.7

Please let me know what other information I can provide.

**Comment by EvanCarroll** [comment]
Having the same issue, I'm getting, 

```
err:   D3D9Query: Unsupported query type 4 (from d3d9.supportVCache)
warn:  ConvertFormat: Unknown format encountered: D3D9Format::AL16
warn:  ConvertFormat: Unknown format encountered: D3D9Format::R16
```

I also see,

`warn:  OpenVR: Failed to locate module`

and 

`warn:  D3D9: VK_FORMAT_D16_UNORM_S8_UINT -> VK_FORMAT_D24_UNORM_S8_UINT`

**Comment by regloff72** [comment]
Same issue here as well. 

Fedora 32
5.8.12-200.fc32.x86_64
nVidia driver version: 450.66

Wine 5.17 
lutris-0.5.7.1-1.fc32.x86_64

It runs pretty well otherwise - but having this exact same issue. Just in some zones on some textures. 

I did note a number of DDS files reported as not found in 'textures.txt', but can't say if that's at all related.

[Fri Oct 02 21:04:35 2020]00004:ResourceManager::CreateTexture(): unable to load "c_ALA_br_s24_n.dds" (not found)
[Fri Oct 02 21:05:23 2020]00005:ResourceManager::CreateTexture(): unable to load "4000GENERICCHAIN_N.DDS" (not found)
[Fri Oct 02 21:05:34 2020]00006:ResourceManager::CreateTexture(): unable to load "c_ALA_bd_s24_c.dds" (not found)

etc.. 

 I see the same format warnings as EvanCarroll gets as well. 

net.lutris.Lutris.desktop[32194]: warn:  ConvertFormat: Unknown format encountered: D3D9Format::AL16
net.lutris.Lutris.desktop[32194]: warn:  ConvertFormat: Unknown format encountered: D3D9Format::R16
net.lutris.Lutris.desktop[32194]: warn:  D3D9: VK_FORMAT_D16_UNORM_S8_UINT -> VK_FORMAT_D24_UNORM_S8_UINT
net.lutris.Lutris.desktop[32194]: warn:  D3D9: VK_FORMAT_A4R4G4B4_UNORM_PACK16_EXT -> VK_FORMAT_B4G4R4A4_UNORM_PACK16

**Comment by rennil42** [comment]
I am experiencing the same black textures in certain newer zones.

Ubuntu 20.04
5.4.0-54-generic x86_64
nVidia driver version: 455.38
Wine 5.0.3 (stable) and Wine 5.21 (staging)

I get the following console warnings:
warn:  ConvertFormat: Unknown format encountered: D3D9Format::AL16
warn:  ConvertFormat: Unknown format encountered: D3D9Format::R16
warn:  ConvertFormat: Unknown format encountered: D3D9Format::AL16
warn:  ConvertFormat: Unknown format encountered: D3D9Format::R16
info:  D3D9DeviceEx::ResetSwapChain:
info:    Requested Presentation Parameters
info:      - Width:              0
info:      - Height:             0
info:      - Format:             D3D9Format::X8R8G8B8
info:      - Auto Depth Stencil: false
info:                  ^ Format: D3D9Format::Unknown
info:      - Windowed:           true
info:  Restoring display mode: 3840x1600@144
info:  Setting display mode: 3840x1600@144

**Comment by Blisto91** [comment]
@K0bin 
Same as https://github.com/doitsujin/dxvk/issues/1825 this seems to have been fixed in 1.9.2
Can reproduce in 1.6.1, 1.9.1 and works in 1.9.2 & 1.10.1
